### ISSUE TITLE
Umbenennung von Klassen

### ISSUE THREAD

**Issue by mbmueller** [openjverein/jverein#480]
Vielleicht sollte man die Klasse wirklich mal umbenennen in Mitgliedstyp. Man kann ja die Datenbank Tabelle beim alten Namen lassen. Für neue JVerein Entwickler ist halt die Historie nicht immer bekannt.

Das gleiche gilt dann auch für Mitgliedskonto welches man in Sollbuchung umbenennen sollte.

Wenn wir mal einen Stand haben wo nicht zuviele gleichzeitig arbeiten könnte ich das mal machen. Es sind dann halt viele Files betroffen.

_Originally posted by @JohannMaierhofer in https://github.com/openjverein/jverein/pull/446#discussion_r1850882194_

**Comment by lenilsas** [comment]
> * schrittweise Klassen umbenennen incl. der getter bei anderen Klassen

Dass ist mit der Refactoring Funktion von Eclipse ohne großen Aufwand möglich, es wird an allen Orten richtig ersetzt.
Ich denke die größte Schwierigkeit sind die Merg Konflikte bei anderen PRs. Ich denke auch, dass bald wieder ein Release passend wäre. Ich würde gerne die Issues mit den Änderungen bei den Rechnungen noch vorher umsetzen, so dass hier eine einmalige Umstellung ist und nicht beim nächsten Release wieder Änderungen daran sind.

**Comment by JohannMaierhofer** [comment]
> Ich denke die größte Schwierigkeit sind die Merg Konflikte bei anderen PRs. Ich denke auch, dass bald wieder ein Release passend wäre. Ich würde gerne die Issues mit den Änderungen bei den Rechnungen noch vorher umsetzen, so dass hier eine einmalige Umstellung ist und nicht beim nächsten Release wieder Änderungen daran sind.

Ja, das mit den Rechnungen sollte komplett in die nächste Release.

**Comment by mbmueller** [comment]
> * ab jetzt keine neuen Feature PRs, nur Bug fixes

Gilt das dann ab jetzt, oder wollen wir einen Termin festlegen, ab dem dann keine PRs zugelassen werden? Bspw. der 01.01.?

**Comment by JohannMaierhofer** [comment]
Das von mir war nur ein Vorschlag, wir sollten das noch genauer besprechen.
Eine neue Version noch im Dezember denke ich macht Sinn. Es gibt ja schon wieder einige neu Features und dann wäre auch die E-Rechnung für alle bis zum 1. Januar 2025 verfügbar.
Neue  Features könnten auch noch Sinn machen. Wie z.B. das Rechnung Feature vervollständigen.
Für weitere Features wäre dann die Frage bis wann sie da sein müssten damit sie auch noch gereviewt werden können. Ein Review dauert ja auch ein paar Wochen und wir haben auch noch einige offen.

Wir müssen aber auch die Entwicklung nicht bremsen. Wenn du also schon an Features arbeitest, können wir die Umbenennung auch weiter verschieben. Sie eilt ja nicht.

**Comment by JohannMaierhofer** [comment]
> > Wir müssen aber auch die Entwicklung nicht bremsen. Wenn du also schon an Features arbeitest, können wir die Umbenennung auch weiter verschieben. Sie eilt ja nicht.
> 
> Arbeite im Moment eben noch an #434. 

Dann sollten wir bis danach warten mit der Umbenennung.

**Comment by tolot27** [comment]
> * schrittweise Klassen umbenennen incl. der getter bei anderen Klassen

> Dann sollten wir bis danach warten mit der Umbenennung.

Wir sollten erfassen, welche Klassen/Packages wir genau umbenennen bzw. verschieben wollen und dann das Refactoring testen und mit der Klasse/dem Package anfangen, welches die wenigsten Änderungen erfordert. So könnten wir schrittweise vorankommen.

**Comment by lenilsas** [comment]
Ich habe damit schon angefangen eine Liste zusammenzustellen.
Durch das RefaktoringTool von Eclipse sehe ich da wenige Gefahr, dass durch Verschiebungen und Umbenennungen Fehler entstehen.

**Comment by tolot27** [comment]
Ich hätte noch einen kleinen Wunsch, den man beim DB Refactoring eventuell mit umsetzen könnte.
Da ich öfter mal über MySQL nachschaue, was so auf der DB passiert, wäre es schön, wenn die Spalte *buchungsklasse* nach der Spalte *buchungsart* stehen würde und nicht am Ende der Tabelle. Das kann man mit folgendem Statement ändern:
```sql
ALTER TABLE `buchung` 
CHANGE COLUMN `buchungsklasse` `buchungsklasse` BIGINT NULL DEFAULT NULL AFTER `buchungsart`;
```
Ich habe auf die Schnelle keine Update###.java gefunden, wo diese Spalte eingefügt wurde. Das müsste dann natürlich auch angepasst werden. Diese Änderung löscht keine Daten und es werden auch foreign keys berücksichtigt. Bei mir hat es mit MySQL 8.0.39 geklappt. Ob es von MariaDB und H2 unterstützt wird, könnte ich noch prüfen.

**Comment by tolot27** [comment]
> * bei den Listen, soll ich das ListView.java nennen oder mit ListeView.java nennen. Also List oder Liste?
> * bei den Detail views mit Detail im Namen oder ohne?

Momentan werden ja beide Varianten verwendet. Ich bevorzuge die englische Schreibweise im Code, d. h. `ListView` und `DetailView`. Da jedoch mit hoher Wahrscheinlichkeit die Nomen davor auf Deutsch bleiben werden, sind wir beim Denglisch. Interessant wird es dann bei Nomen, wo bereits "liste" drin vor kommt, wie z. B. Anlagenliste. Das ist jetzt vielleicht nicht das beste Beispiel und vielleicht haben wir auch gar keine derartigen Fälle. Aber würde die Klasse `AnlagenlisteListView`, `AnlagenListView` oder nur so wie jetzt bereits `AnlagenlisteView` heißen? Oder Grundsatzfrage: Haben wir/werden wir haben nur `ListView` und `DetailView` oder auch andere Views wie `PrintPreviewView`, `FormView`, etc.?

**Comment by tolot27** [comment]
> > Ich wollte als nächstes die Views umbenennen. Dazu habe ich zwei Fragen :
> > 
> > * bei den Listen, soll ich das ListView.java nennen oder mit ListeView.java nennen. Also List oder Liste?
> > * bei den Detail views mit Detail im Namen oder ohne?
> 
> Ich wäre für Liste und Detail.

Wollen wir dann `View` durch `Ansicht` ersetzen? Dann wäre das Denglisch raus.

**Comment by lenilsas** [comment]
> Wollen wir dann `View` durch `Ansicht` ersetzen? Dann wäre das Denglisch raus.

Ich finde `View` hier schon richtig, da es ja auch die `AbstractView` implementiert. Sonst wird es verwirrend.

**Comment by JohannMaierhofer** [comment]
> > * bei den Listen, soll ich das ListView.java nennen oder mit ListeView.java nennen. Also List oder Liste?
> > * bei den Detail views mit Detail im Namen oder ohne?
> 
> Momentan werden ja beide Varianten verwendet. Ich bevorzuge die englische Schreibweise im Code, d. h. `ListView` und `DetailView`. Da jedoch mit hoher Wahrscheinlichkeit die Nomen davor auf Deutsch bleiben werden, sind wir beim Denglisch. Interessant wird es dann bei Nomen, wo bereits "liste" drin vor kommt, wie z. B. Anlagenliste. Das ist jetzt vielleicht nicht das beste Beispiel und vielleicht haben wir auch gar keine derartigen Fälle. Aber würde die Klasse `AnlagenlisteListView`, `AnlagenListView` oder nur so wie jetzt bereits `AnlagenlisteView` heißen? Oder Grundsatzfrage: Haben wir/werden wir haben nur `ListView` und `DetailView` oder auch andere Views wie `PrintPreviewView`, `FormView`, etc.?

Da stellt sich die Frage ob man bei den ListView.java vorne die Einzahl oder Mehrzahl benutzt. Es kommt ja beides aktuell vor. Also AnlageListView oder AnlagenListView.
Bei Mehrzahl wirds interessant, z.B. ProjekteListView oder ProjektListView. 

AnlagenlisteView denke ich nicht, weil man das ja sonst auch machen könnte, also ProjektlisteView etc. Oder halt dann überall so.

**Comment by JohannMaierhofer** [comment]
OK, ich mache das dann Singular und mit ListeView und Detail.
